src/components/admin/EditDialogue.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useContext } from "react";
+import { useForm, type SubmitHandler } from "react-hook-form";
+
+import { TextInput } from "@/components/TextInput";
+import { Button } from "@/components/ui/button";
+import { gameSessionSchedules } from "@/lib/db/schema";
+import {
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "../ui/dialog";
+import { ScheduleContext } from "./SemesterDetailCard";
+
+type UserResponse = {
```

Comment:
User Response is really vague, can you rename this to something more specific? Like EditSemesterFormData